### PR TITLE
Use a Rtree query for tagMesh*Round 

### PR DESCRIPTION
**Pull request by Summerdave**
This PR targets issue #293 by using boost::geometry for lookup.

### CODE REVIEW

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 1]
```diff
@@ -770,24 +770,25 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshFirstRound()
...
+          }));
     #else
       #warning "Mesh filtering deactivated, due to PETSc version < 3.8. \
 preCICE is fully functional, but performance for large cases is degraded."
     #endif
-    if (isInside)
-      v.tag();
+  }
+  for (auto& vert : insideVerts){
+    vert.tag();
```
Comment (floli) [inline src/mapping/PetRadialBasisFctMapping.hpp]: Idea:

Move that loop (`for vert in insideVerts: vert.tag()`) into `if (not _basisFunction.hasCompactSupport)` and use the `make_function_output_iterator` to directly invoke `vert.tag()`. Got the idea?

File: src/mapping/PetRadialBasisFctMapping.hpp [hunk 2]
```diff
@@ -825,18 +826,20 @@ void PetRadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::tagMeshSecondRound()
...
+    bb[d].second += _basisFunction.getSupportRadius();
+  }
+  auto rtree = mesh::rtree::getVertexRTree(mesh);
+  mesh::Mesh::VertexContainer insideVerts;
+  rtree->query(boost::geometry::index::within(bb),
+      boost::make_function_output_iterator([&mesh, &insideVerts](size_t idx){
+        insideVerts.push_back(&mesh->vertices()[idx]);
+      }));
+  for (auto& vert : insideVerts){
+    vert.tag();
```
Comment (floli) [inline src/mapping/PetRadialBasisFctMapping.hpp]: Same may be possible here.

File: src/mapping/PetRadialBasisFctMapping.hpp (hunk 1 above)
Comment (Summerdave) [reply src/mapping/PetRadialBasisFctMapping.hpp]: Clever.

**Comment by fsimonis** [comment]
@Summerdave Looks good.

Can you implement the tagging for the other mappings too? (The title of the PR suggests you change all of them.)

_For the record: I don't like the repetition of the type definition for Mesh::BoundingBox, but I get why you did it. Maybe we could add something like a `mesh_fwd.hpp` in the future?_

**Comment by Summerdave** [comment]
> Can you implement the tagging for the other mappings too? (The title of the PR suggests you change all of them.)

The re-partitioning process for RBF mapping is based on @uekerman`s [thesis](https://mediatum.ub.tum.de/doc/1320661/document.pdf) (page 89). Right now, the partitioning for the other mappings works by computing a mapping and filtering this by whether the point _was used_. So for determining which points to tag a full mapping must be run anyway. @uekerman has also commented on this in #293.

**Comment by Summerdave** [comment]
@floli Found a severe issue in the code which is now fixed. I also added unit tests to ensure `tagMeshFirstRound` and `tagMeshSecondRound` work correctly.

**Pull request merged**